### PR TITLE
Devenv - Vagrant dev environment provisioned by fabric

### PR DESCRIPTION
**Pull request by primozs**
I would propose to replace chef vagrant provisioning with python fabric.
To make it even more unified i would suggest to make also new fabfile package module and put all fabric related stuff there. Currently i created "vagrant provision" task in a separate fabric file.
What do you think?
Cheers

### CODE REVIEW

**Comment by Turbo87** [comment]
Getting rid of Chef/Puppet sounds like a good idea to me. I wasn't actually aware that fabric could be used that easily for stuff like that...

**Comment by primozs** [comment]
Hi Turbo, 
Yes fabtools and cuisine are providing some nice functions for provisioning similar to chef. 
I have updated the script and install instructions.

**Comment by fb** [comment]
@primozs: I tried setting up a VM with your branch and it worked nearly without problems. I just had to ensure apt-get update is run in the VM before provisioning. Otherwise it would fail because of outdated package URLs. Please merge my branch pr-370 to your pull request.

**Comment by primozs** [comment]
@fb Hi, I did not notice any problems on my side but you are right to run "update", I have merged your branch.

File: Vagrantfile [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-
-  config.vm.provision :chef_solo do |chef|
-    chef.cookbooks_path = [".cookbooks", "cookbooks"]
-    
-    chef.add_recipe "skylines"
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 512
+    v.cpus = 1
+    v.gui = true
```
Comment (Turbo87) [inline Vagrantfile]: do we need the GUI?

**Comment by primozs** [comment]
@Turbo87 I like it so i dont forget to turn off the machine after i stop working. but we can remove it also. What do you think?

**Comment by primozs** [comment]
Would it make any sense to add also vagrant deploy test setup similar like for production?

File: Vagrantfile (hunk 1 above)
Comment (Turbo87) [inline Vagrantfile]: afaik it's using more resources, so I'm not a big fan of it if it's not strictly needed

Comment (primozs) [inline Vagrantfile]: I have removed the gui flag

**Comment by primozs** [comment]
Don't merge this, i have some other version i would propose in a few days.

**Comment by Turbo87** [comment]
@primozs any news on that other environment?

**Comment by primozs** [comment]
@Turbo87 Hi, I have something in a separate project. I would need to merge and clean it. Maybe I could do it after tomorrow.

**Comment by Turbo87** [comment]
@primozs I've replaced it with a simple "shell" provisioning for now (see #388), but feel free to improve it

**Pull request closed without merging**